### PR TITLE
Junos: routing-instances resource module added

### PR DESCRIPTION
**Pull request by rohitthakur2590**
##### SUMMARY

Depends-On: https://github.com/ansible-collections/junipernetworks.junos/pull/164
resolve: https://github.com/ansible-collections/junipernetworks.junos/issues/117

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/junos_routing_instances.py [hunk 1]
```diff
@@ -0,0 +1,765 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
```
Comment (justjais) [inline plugins/modules/junos_routing_instances.py:3]:
```suggestion
# Copyright 2021 Red Hat
```

File: plugins/modules/junos_routing_instances.py [hunk 2]
```diff
@@ -0,0 +1,765 @@
...
+      - name: "forwardinst"
+        type: "forwarding"
+        description: "Configured by Ansible Content Team"
+    state: merged
+#
+# -------------------------
+# Module Execution Result
+# -------------------------
+#
+# After state
```
Comment (justjais) [inline plugins/modules/junos_routing_instances.py:214]: @rohitthakur2590 can you plz include the cmd ran in the examples as well

File: plugins/modules/junos_routing_instances.py [hunk 3]
```diff
@@ -0,0 +1,765 @@
...
+# -----------
+#
+# admin@vsrx# show routing-instances
+#
+# [edit]
+
+- name: Gather Junos BGP address family config
+  junipernetworks.junos.junos_routing_instances:
+    config:
```
Comment (justjais) [inline plugins/modules/junos_routing_instances.py:485]:
```suggestion
    
```

File: plugins/modules/junos_routing_instances.py [hunk 4]
```diff
@@ -0,0 +1,765 @@
...
+# After state
+# -----------
+#
+# admin@vsrx# show routing-instances
+# forwardinst {
+#     description "Configured by Ansible Content Team";
+#     instance-type forwarding;
+# }
+
+# Using deleted
```
Comment (justjais) [inline plugins/modules/junos_routing_instances.py:448]:
```suggestion
# Using deleted without config
```

File: tests/unit/modules/network/junos/test_junos_routing_instances.py [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+            "<nc:vrf-export>test-policy</nc:vrf-export><nc:vrf-export>test-policy-1</nc:vrf-export>"
+            "</nc:instance></nc:routing-instances>"
+        ]
+        result = self.execute_module(changed=True, commands=commands)
+        self.assertEqual(sorted(result["commands"]), sorted(commands))
+
+    def test_junos_routing_instances_merged_idempotent(self):
+        set_module_args(
+            dict(
+                config=[
```
Comment (justjais) [inline tests/unit/modules/network/junos/test_junos_routing_instances.py:134]: @rohitthakur2590 have u covered all of the module params accepted by the module?

Comment (rohitthakur2590) [reply tests/unit/modules/network/junos/test_junos_routing_instances.py:134]: big xml trees are generated, if required I'll be updating that in different PR

File: plugins/modules/junos_routing_instances.py (hunk 2 above)
Comment (rohitthakur2590) [reply plugins/modules/junos_routing_instances.py:213]: Big netconf based xml trees got generated, if required I can add that in different PR.

File: tests/unit/modules/network/junos/test_junos_routing_instances.py (hunk 5 above)
Comment (rohitthakur2590) [reply tests/unit/modules/network/junos/test_junos_routing_instances.py:134]: yes , its covered between units and integration tests

**Review by justjais — COMMENTED** [review]
@rohitthakur2590 changes LGTM, minor cosmetic comment is to update the year in module_utils files, it seems I missed it the last time

**Review by justjais — APPROVED** [review]

**Pull request merged**